src/plugins/intel_gna/src/request/worker_impl.cpp

```diff
@@ -42,23 +43,36 @@ Gna2Model* WorkerImpl::model() {
 void WorkerImpl::enqueueRequest() {
     check_if_free();
 
-    for (auto& subrequest : modelSubrequests_) {
-        subrequest->enqueue();
+    try {
+        for (auto& subrequest : modelSubrequests_) {
+            subrequest->enqueue();
+        }
+    } catch (const std::exception& e) {
+        ov::intel_gna::log::error() << "Exception when enqueueing inference: " << e.what() << std::endl;
+        cleanup_unfinished_submodules();
+        throw;
     }
 }
 
 RequestStatus WorkerImpl::wait(int64_t timeoutMilliseconds) {
     bool pending = false;
 
-    // iterate over all configurations for requst
-    for (auto& subrequest : modelSubrequests_) {
-        if (!subrequest->isPending()) {
-            continue;
-        }
+    try {
+        // iterate over all configurations for requst
+        for (auto& subrequest : modelSubrequests_) {
+            if (!subrequest->isPending()) {
+                continue;
+            }
 
-        if (subrequest->wait(timeoutMilliseconds) == RequestStatus::kPending) {
-            pending = true;
+            if (subrequest->wait(timeoutMilliseconds) == RequestStatus::kPending) {
+                pending = true;
+            }
         }
+    } catch (const std::exception& e) {
+        ov::intel_gna::log::error() << "Exception when executiong wait: " << e.what() << std::endl;
+        pending = false;
+        cleanup_unfinished_submodules();
+        throw;
     }
```

Comment:
1. i would get rid of try/catch and introduce RequestStatus::kCompletedWithError
2. cleanup_unfinished_submodules() would be not needed
3. We dont need to throw from this function, we can return kCompletedWithError and throw from outside function

Reply:
1. Ok
2. We still  need try to cleanup due the fact we could receive kPending.
3. Ok

Reply:
Ad2. 
make sense,
...
however if we get booth: pending & kCompletedWithError (for subrequests) we could just return pending (overal status) and do the de-facto cleanup at the next call to wait and throw kCompletedWithError when all pending subrequests are completed,

this scenario is not likely to happen (combines three unusual situations big model + pending +error from wait) so making the code simpler is IMO the most important

Reply:
I have changed worker to not throw for enqueue and wait, but that required some changes in gna_plugin.cpp